### ISSUE TITLE
`gitihub open [url]` should display a helpful error message

### ISSUE THREAD

**Issue by shiftkey** [desktop/desktop#4658]
**Is your feature request related to a problem? Please describe.**

There was a discussion in #4259 about the friction of `github open` displaying a dialog to confirm adding a repository, and it was agreed that this was unnecessary. #4513 has been merged, but there was an unresolved question about how we handle URLs.

**Describe the solution you'd like**

I'd like `github open [url]` to be a bit more paranoid about "things that looks like remote URLs" and defensively indicate what needs to be done.

Something like this:

```
$ github open https://github.com/desktop/desktop

You cannot open a remote URL in GitHub Desktop
Use `git clone https://github.com/desktop/desktop` instead to initiate the clone
```

And the `github` CLI should return an error code here.

**Describe alternatives you've considered**

The current behaviour is to consider the URL to be a relative path, and open the dialog to that "path", which doesn't make much sense.

<img width="471" src="https://user-images.githubusercontent.com/359239/39900861-18edb246-5508-11e8-8779-15f87de78a8a.png">

I'm also :-1: on implicitly redirecting this to `github clone [url]` - this might also be confusing.

**Teachability, Documentation, Adoption, Migration Strategy**

Not sure if there's anything else to cover off here 🤷‍♂️

**Comment by msztech** [comment]
I think we could show the error message directly in the command line if a user enters `github open` with an URL rather than a path. We can detect url by checking if it's prefixed with https:// for one type of repo address. But what if the user enters `git@github.com:desktop/desktop.git`? Maybe check the suffix `.git`? Do you think this will work? @shiftkey

**Comment by shiftkey** [comment]
@msztech apologies for the delay in replying. 

We have existing logic for this that could be used here, rather than hand-rolling any new checks:

https://github.com/desktop/desktop/blob/0ed30be2cece05603ff4f10d9b7c4be241299a7d/app/src/lib/remote-parsing.ts#L47-L62

**Comment by msztech** [comment]
Hi! ;)

Then I can work on this. I’ll call the remote parse to see it’s a remote path, if so, show the error message in CLI or GUI.

**Comment by msztech** [comment]
Working on this. Will PR ASAP @shiftkey

**Comment by msztech** [comment]
@shiftkey Pull Requested. ;)

**Comment by msztech** [comment]
Also, will it be better if we show ‘github clone’ instead of ‘git clone’ as the prompt since that will open up the dialog instead of cloning the remote to the current path? @swiftkey

**Comment by shiftkey** [comment]
This was resolved in #7452 and published as part of Desktop 2.0. Thanks again @msztech!
